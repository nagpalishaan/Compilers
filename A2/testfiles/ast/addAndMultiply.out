Program
  List
    Function
      IdDecl ID="result"
      Opt
      List
        DeclareVar
          IdDecl ID="a"
          Opt
        DeclareVar
          IdDecl ID="b"
          Opt
        AssignVar
          IdUse ID="a"
          Numeral NUMERAL="2"
        AssignVar
          IdUse ID="b"
          Numeral NUMERAL="3"
        AssignVar
          IdUse ID="c"
          Add
            IdUse ID="a"
            Mul
              IdUse ID="b"
              Numeral NUMERAL="2"
        AssignVar
          IdUse ID="d"
          Add
            Mul
              Numeral NUMERAL="2"
              IdUse ID="a"
            IdUse ID="b"
