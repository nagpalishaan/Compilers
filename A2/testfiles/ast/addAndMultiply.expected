Program
  List
    Function
      IdDecl ID="result"
      Opt
      List
        DeclareVar
          IdDecl ID="a"
          Opt
        DeclareVar
          IdDecl ID="b"
          Opt
        AssignVar
          IdDecl ID="a"
          Numeral NUMERAL="2"
        AssignVar
          IdDecl ID="b"
          Numeral NUMERAL="3"
        AssignVar
          IdDecl ID="c"
          Add
            IdUse ID="a"
            Mul
              IdUse ID="b"
              Numeral NUMERAL="2"
        AssignVar
          IdDecl ID="d"
          Add
            Mul
              Numeral NUMERAL="2"
              IdUse ID="a"
            IdUse ID="b"
